src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -333,19 +333,33 @@ void java_bytecode_parsert::get_class_refs()
     }
   }
 
-  for(const auto &m : parse_tree.parsed_class.fields)
+  for(const auto &f : parse_tree.parsed_class.fields)
   {
-    typet t=java_type_from_string(m.signature);
+    typet t;
+    if(f.has_signature)
+    {
+      t=java_type_from_string(f.signature);
+    }
+    else
+      t=java_type_from_string(f.descriptor);
```

Comment:
The fact that the `if` statement has braces around its body, but else doesn't, and is also followed by another statement immediately after its body. This is inviting bugs and confusion in the future, I think.